locator/tablets.cc

```diff
@@ -245,9 +245,48 @@ dht::token_range tablet_map::get_token_range(tablet_id id) const {
     }
 }
 
-host_id tablet_map::get_primary_replica(tablet_id id) const {
-    const auto info = get_tablet_info(id);
-    return info.replicas.at(size_t(id) % info.replicas.size()).host;
+future<tablet_map::primary_replica_map> tablet_map::calc_primary_replica_map(table_id table, const topology& topo, sstring dc) const {
```

Comment:
Please use normal words like calculate.

Reply:
Is this function stable? Will it return the same result on all nodes? How much does a single tablet migration affect it? If this is guaranteed, why is it not documented?

Reply:
The function is stable and will return the same result on all nodes assuming that the topology is the same on all nodes. (and we already know we have to keep it stable throught repair so not to miss repairing a tablet if it migrated under our feet)

That said, the load-balanced selection approach might be more sensitive to single tablet migration since might cause a ripple effect due to difrrent primary selection at this tablet which changes the cumulative load map and therefore affects further selections from that point on (since the per-tablet selections are no longer independent of each other as was the case with the previous algorithm I suggested that was more sensitive to bad tablet ownership imbalance.